### PR TITLE
📖 Added beginner's guide to scorecard checks docs

### PR DESCRIPTION
**Pull request by ariathaker**
-Also edited README to link to the beginner's guide

#### What kind of change does this PR introduce?

Docs update 

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

N/A

#### What is the new behavior (if this is a feature change)?**

N/A

#### Which issue(s) this PR fixes

Addresses issue #3434 --[Docs addition: getting started guide](https://github.com/ossf/scorecard/issues/3434).  

#### Special notes for your reviewer

N/A

#### Does this PR introduce a user-facing change?

```
NONE

```

### CODE REVIEW

File: docs/beginner-checks.md
Comment (spencerschrock) [inline docs/beginner-checks.md:3]: Should the OpenSSF link go to the securityscorecards page? or https://openssf.org/ ?

File: docs/beginner-checks.md
Comment (spencerschrock) [inline docs/beginner-checks.md:24]: I think Scorecard looks for it at the project's root by default

File: docs/beginner-checks.md
Comment (spencerschrock) [inline docs/beginner-checks.md:42]: this is missing an indentation for `contents`

File: docs/beginner-checks.md
Comment (spencerschrock) [inline docs/beginner-checks.md:36]: Probably worth mentioning that this is for the top level permissions. Some workflows still might require these `write` permissions, the goal isn't to eliminate all of them. Instead frame this as a matter of least privilege?

File: docs/beginner-checks.md
Comment (spencerschrock) [inline docs/beginner-checks.md:56]: note: the `/en/` part of the link forces the english language doc, instead of auto-detecting the user's locale.

Prefer removing it when possible:
https://docs.github.com/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks

Comment (spencerschrock) [reply docs/beginner-checks.md:56]: Might be a bit of a nit, but the required status checks don't ensure that all tests pass. Rather it only enforces the ones the maintainer chooses

File: docs/beginner-checks.md
Comment (spencerschrock) [inline docs/beginner-checks.md:57]: Signed commits aren't actually judged by scorecard, should we be recommending them here?

File: docs/beginner-checks.md
Comment (ariathaker) [reply docs/beginner-checks.md]: Thanks, that makes sense, I changed the link.

File: docs/beginner-checks.md
Comment (ariathaker) [reply docs/beginner-checks.md]: Thanks! Changed the text to reflect that.

File: docs/beginner-checks.md
Comment (ariathaker) [reply docs/beginner-checks.md]: Thanks, added

File: docs/beginner-checks.md
Comment (ariathaker) [reply docs/beginner-checks.md]: Thanks, that makes sense. I've revised the paragraph to reflect your note.

File: docs/beginner-checks.md
Comment (ariathaker) [reply docs/beginner-checks.md]: I definitely see what you mean. I've removed the bullet point about signed commits.

File: docs/beginner-checks.md
Comment (olivekl) [inline docs/beginner-checks.md:68]:
```suggestion
Another benefit to releasing projects as packages is reproducibility—the version that new users can download and execute is identical to the one that you and other contributors have already reviewed. Packages also have clear versioning documentation that makes it easier to track whether any newly discovered security issues are applicable to your project.
```

File: docs/beginner-checks.md
Comment (olivekl) [inline docs/beginner-checks.md:72]:
```suggestion
Based on the specifics of your project, not all the checks offered by Scorecard or discussed on this page may apply to you. For example, if you are the sole maintainer of an open-source project, the “Code Review” check would not be usable for your project.
```

**Review by olivekl — COMMENTED** [review]
Thank you for this fantastic guide, @ariathaker! Left some comments for clarity / tone, but overall LGTM --- this will be a great resource for maintainers new to this area.

File: docs/beginner-checks.md
Comment (hayleycd) [reply docs/beginner-checks.md:3]: Agree, now that it would actually be in the scorecard docs, readers wouldn't need this background.

**Comment by olivekl** [comment]
Thanks for the updates, @ariathaker! It looks like you'll need to [sign a DCO](https://github.com/ossf/scorecard/pull/3617/checks?check_run_id=18262763588) to get this merged.

@spencerschrock What're the next steps for review here?

**Comment by ariathaker** [comment]
Ah, thanks @olivekl! I just rebased to include the signature.

**Comment by spencerschrock** [comment]
> Thanks for the updates, @ariathaker! It looks like you'll need to [sign a DCO](https://github.com/ossf/scorecard/pull/3617/checks?check_run_id=18262763588) to get this merged.
> 
> @spencerschrock What're the next steps for review here?

I'll take a look again tomorrow 👀.

File: docs/beginner-checks.md
Comment (spencerschrock) [inline docs/beginner-checks.md:3]: might be nice to make this list act as a mini table of contents, and link to the three sections?

File: docs/beginner-checks.md
Comment (spencerschrock) [inline docs/beginner-checks.md:34]: > Projects that have top-level tokens set to `write` access for their automated workflows may be vulnerable to malicious code execution

I think this isn't quite the whole story, the "may" is doing a lot of heavy lifting here. Perhaps we should rephrase? I haven't found a succinct way yet of what I'm intending, but something like:

> Projects that have don't follow least privilege, particularly around `write` permissions, are putting more trust than necessary in their workflow dependencies, introducing an attack vector if those dependencies are ever compromised.

File: docs/beginner-checks.md
Comment (spencerschrock) [inline docs/beginner-checks.md:66]: spaces needed around the dash here?
> reproducibility—the

File: docs/beginner-checks.md
Comment (spencerschrock) [inline docs/beginner-checks.md:68]: should this disclaimer section be part of the top paragraph?

**Comment by spencerschrock** [comment]
You should also change the PR from draft to "ready for review"

File: docs/beginner-checks.md
Comment (spencerschrock) [reply docs/beginner-checks.md]: I think this is resolved?

File: docs/beginner-checks.md
Comment (spencerschrock) [reply docs/beginner-checks.md]: The links will need fixed slightly, for example for the first one it's looking for:
`#1-setting-up-your-project`. (you can see this by previewing the file and clicking on the "link" button for the section)

File: docs/beginner-checks.md
Comment (olivekl) [reply docs/beginner-checks.md]: Not necessary! An em dash usually doesn't have spaces, though some style guides differ. If you'd prefer spaces, though, that's fine :)

File: docs/beginner-checks.md
Comment (olivekl) [reply docs/beginner-checks.md]: I think the suggested version might not be explicit enough for the intended audience. Would you be open to something finding a balance between the two, like:

>Projects that have top-level tokens set to `write` are granting more access than necessary to their automated workflow dependencies. If those dependencies were ever compromised, they could become an attack vector, and in the worse case could be vulnerable to malicious code execution.

File: docs/beginner-checks.md
Comment (spencerschrock) [reply docs/beginner-checks.md]: just looked like a typo to me haha. will defer to you

Comment (olivekl) [reply docs/beginner-checks.md]: For total correctness, let's have @ariathaker just double check that it's entered as an em dash and not an en dash or hyphen. I think it should look fine in the actual README, though. :)

**Review by ariathaker — COMMENTED** [review]
Very sorry, I didn't realize the "Pending" badge meant that my comments and changes weren't visible! Submitting them all now

File: docs/beginner-checks.md
Comment (ariathaker) [reply docs/beginner-checks.md]: Yes, sorry, I'd replied and made the cut but didn't realize "Pending" meant my comment wasn't visible

File: docs/beginner-checks.md
Comment (ariathaker) [reply docs/beginner-checks.md]: Just verified that it's an em dash!

File: docs/beginner-checks.md
Comment (ariathaker) [reply docs/beginner-checks.md]: I think @olivekl's suggestion works well since it really emphasizes the potential consequences of not following least privilege. Have made the change for now but of course am happy to revert to the previous version or to go with any other revisions @spencerschrock suggests

File: docs/beginner-checks.md
Comment (spencerschrock) [reply docs/beginner-checks.md:66]: Sorry, this comment was specifically for the "## Customize your checks to your project’s needs" part. Not the "Another benefit" part.

I forget GitHub can be misleading when leaving comments on single lines, sorry.

Comment (spencerschrock) [reply docs/beginner-checks.md:66]: Specifically, I thought this section could be useful at the very top of the page somewhere.

> Based on the specifics of your project, not all the checks offered by Scorecard or discussed on this page may apply to you.

Maybe just the one sentence with a link to the full section?

**Comment by spencerschrock** [comment]
> Very sorry, I didn't realize the "Pending" badge meant that my comments and changes weren't visible! Submitting them all now

It's happened to me before too. In my experience, it depends on which screen you use to leave the comments.

File: docs/beginner-checks.md
Comment (ariathaker) [reply docs/beginner-checks.md:70]: Got it, made the change. Thanks!

**Review by spencerschrock — APPROVED** [review]
Thanks!

**Pull request merged**
